### DISCUSSION TITLE
Support for mixed-bag (or known as lumped) models to calculate biomass fluxes of each taxon?

### DISCUSSION THREAD

**Discussion by hellopeccat** [micom-dev/micom#195] [Q&A] [answered]
Dear MICOM team, dear community,

Pretty glad to learn MICOM. I have got some community-scale models generated by a mixed-bag function annotation directly from contigs in advance, but I am going to binning those contigs into MAGs to get the abundance of each taxon, together with mapping MAGs to public model databases to deliever a specific biomass reaction to each taxon. In this case, could I use MICOM to calculate biomass fluxes of each taxon directly from the set of biomass reactions integrated to my original mixed-bag models? (which means that except for setting various biomass production, we would consider the community as a single supra organism.)

I am also considering to firstly annotate functions of each taxon, subsequently build individual models for each taxon, and finally merge them into a compartmentalized model. However it seems to be much time-cost for the separate annotation than mixed-bag annotation, since there would be 300+ species in each sample. Directly use the matched public models may be an alternative solution, but for some reasons I prefer to use my custom counterparts. **The long run time is the most essential reason that I would like to learn whether the strategy of ctFBA is available for mixed-bag models in silico computation.**  

Thanks for any comment.

**Comment by cdiener** [comment, marked as answer]
Hi,

MICOM does not implement mixed-bag models at the moment. You are right though that in theory you could apply ctFBA to a mixed-bag model to solve the multi-objective problem there. You could do it with cobrapy for a mixed-bag model if you really wanted to by manually doing the two ctFBA steps. You would just have to keep a list with the individual and community-wide biomass reactions around.

The second strategy is what we are often using those days - models built from MAGs and then use those in the community models. I agree that it is indeed more time-consuming but you do get a bit more realistic models out of it (where taxa don't share ATP between each other for instance).

**Comment by hellopeccat** [comment]
Hello,

I feel excited with your positive comments though I could not directly finish this sort of job in the very near future. I will try it when I get more free time. Thank you for your valuable and constructive reply. 

May I ask another little question here? Not much relevant with this discussion. Plentiful studies on gut microbiomes were based on AGORA, truly providing many convenience, however it seems that no comparable databases curate metabolic models for the diverse microbes living in other habitats. Right? Can we conclude that the current status is partially resulted from the less divergent 'media'(diet) of gut, compared to the 'media'(nutrient composition) of other nature environments? For example, average western diet was used as the medium for gut to set flux bound, while it is difficult to get an average medium for the environments in many scenarios.
